numpy/_core/src/multiarray/item_selection.c

```diff
@@ -1067,7 +1069,18 @@ PyArray_Choose(PyArrayObject *ip, PyObject *op, PyArrayObject *out,
             flags |= NPY_ARRAY_ENSURECOPY;
         }
         Py_INCREF(dtype);
-        obj = (PyArrayObject *)PyArray_FromArray(out, dtype, flags);
+        if (!PyArray_EquivTypes(dtype, PyArray_DESCR(out))) {
+            obj = (PyArrayObject *)PyArray_NewFromDescr(Py_TYPE(out),
```

Comment:
I don't think there is any reason to use `Py_TYPE(out)` here, just pass `NULL` and get a base-class array, I think.

(There _is_ a subtle issue here about that, I think, in that ufuncs would ensure they call `__array_wrap__` when done and we don't.  But this a completely different issue!)

Reply:
Makes sense thanks, but looking at the doc for `PyArray_NewFromDescr`, should we pass `&PyArray_Type` instead? `NULL` causes a `ValueError`.

Reply:
Yes, of course.  Somehow thought it accepted NULL as PyArray_Type there.

Reply:
I made the change but it seems to fail a regression test `test_attributes` that tests for an identical `info` string assigned to the out array.

Reply:
Maybe push, but may not look at it today. I really don't see why it matters since `obj` is never returned to the user!  (And not understanding that worries me more than the change ;)).

Reply:
No problem at all, thanks for the feedback. Yes, it is a bit confusing...

Reply:
Oops, I think I was editing the wrong line! Things work fine now, pushed. Thanks.